Assets/Scripts/Tools/ExtendedRandom.cs

```diff
@@ -0,0 +1,195 @@
+        /// </summary>
+        public Vector2 InsideUnitCircle()
+        {
+            float angle = Value(0, 2) * Mathf.PI; // Generate angle between [0, 2*Pi)
+            float radius = Value(); // Generate radius between [0, 1)
+
+            // Convert to from polar coordinates to regular coordinates
+            return new Vector2(
+                x: radius * Mathf.Cos(angle),
+                y: radius * Mathf.Sin(angle));
+        }
+
+        /// <summary>
+        /// <para>Returns a uniformly distributed random point inside or on a circle with radius 1.0</para>
+        /// <i>Note: This function is slower than <see cref="InsideUnitCircle()">InsideUnitCircle()</see></i>
+        /// </summary>
+        public Vector2 InsideUnitCircleUniform()
+        {
+            float angle = Value(0, 2) * Mathf.PI; // Generate angle between [0, 2*Pi)
+            float radius = Mathf.Sqrt(Value()); // Generate radius between [0, 1)
+
+            // Convert to from polar coordinates to regular coordinates
+            return new Vector2(
+                x: radius * Mathf.Cos(angle),
+                y: radius * Mathf.Sin(angle));
+        }
+
+        /// <summary>
+        /// Returns a uniformly random point on the edge of a circle with radius 1.0
+        /// </summary>
+        public Vector2 OnUnitCircle()
+        {
+            float angle = Value(0, 2) * Mathf.PI; // Generate angle between [0, 2*Pi)
+
+            // Convert to from polar coordinates to regular coordinates
+            return new Vector2(
+                x: Mathf.Cos(angle),
+                y: Mathf.Sin(angle));
+        }
+
+        /// <summary>
+        /// <para>Returns a random point inside or on a sphere with radius 1.0</para>
+        /// <i>Disclaimer: This function is not uniformly distributed</i> 
+        /// </summary>
+        public Vector3 InsideUnitSphere()
+        {
+            float phi = Value(0, 2) * Mathf.PI; // Generates angle between [0, 2*Pi)
+            float theta = Value() * Mathf.PI; // Generates angle between [0, Pi)
+            float radius = Value(); // Generates radius between [0, 1]
+
+            // Convert from spherical coordinates to regular coordinates
+            return new Vector3(
+                x: radius * Mathf.Sin(theta) * Mathf.Cos(phi),
+                y: radius * Mathf.Sin(theta) * Mathf.Sin(phi),
+                z: radius * Mathf.Cos(theta));
+        }
+
+        /// <summary>
+        /// <para>Returns a uniformly distributed random point inside or on a sphere with radius 1.0</para>
+        /// <i>Note: This function is slower than <see cref="InsideUnitSphere()">InsideUnitSphere()</see></i>
+        /// </summary>
+        public Vector3 InsideUnitSphereUniform()
+        {
+            float phi = Mathf.Acos(Value(0, 2) - 1); // Generates angle between [0, 2*Pi)
+            float theta = Value(0, 2) * Mathf.PI; // Generates angle between [0, Pi)
```

Comment:
Descriptions seem switched. First is 0-Pi and second is 0-2Pi

Reply:
fixed